File: Frontend/src/app/model/car.ts
import { UserInfo } from './userInfo';
import { Model } from './model';
import { Make } from './make';
import { BodyStyle } from 'src/app/model/bodyStyle';
import { GearBoxType } from './gearboxType';
import { FuelType } from './fuelType';

export class Car {
    id: number;
    make: Make;
    model: Model;
    fuelType: FuelType;
    gearBoxType: GearBoxType;
    bodyStyle: BodyStyle;
    mileageInKm: number;
    kidsSeats: number;
    availableTracking: boolean;
    avgRating: number;
    owner: UserInfo;
    commentsCount: number;

    constructor(make: Make, model: Model, fuelType: FuelType, gearBoxType: GearBoxType, bodyStyle: BodyStyle,
        mileageInKm: number, kidsSeats: number, availableTracking: boolean, id?: number, avgRating?: number, owner?: UserInfo, commentsCount?: number) {
        this.make = make;
        this.model = model;
        this.fuelType = fuelType;
        this.gearBoxType = gearBoxType;
        this.bodyStyle = bodyStyle;
        this.mileageInKm = mileageInKm;
        this.kidsSeats = kidsSeats;
        this.availableTracking = availableTracking;
        this.id = id;
        this.avgRating = avgRating;
        this.owner = owner;
        this.commentsCount = commentsCount;
    }

}

File: MainApp/SearchService/src/main/java/jvn/SearchService/model/Car.java
package jvn.SearchService.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jvn.SearchService.enumeration.LogicalStatus;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.HashSet;
import java.util.Set;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Car {
    @Id
    private Long id;

    @Column(nullable = false)
    private LogicalStatus logicalStatus = LogicalStatus.EXISTING;

    @Column(nullable = false)
    private String make;

    @Column(nullable = false)
    private String model;

    @Column(nullable = false)
    private String fuelType;

    @Column(nullable = false)
    private String gearBoxType;

    @Column(nullable = false)
    private String bodyStyle;

    @Column(nullable = false)
    private Integer mileageInKm;

    @Column(nullable = false)
    private Integer kidsSeats;

    @ElementCollection
    private Set<String> pictures = new HashSet<>();

    @Column(nullable = false)
    private Boolean availableTracking;

    @Column(nullable = false)
    private Long owner;

    @JsonIgnore
    @OneToMany(mappedBy = "car", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private Set<Advertisement> advertisements = new HashSet<>();

    @Column
    private Double avgRating;

    @Column
    private Integer commentsCount;
}
